<launch>
    <arg name="camera_name" default="front_camera" />
    <arg name="camera_frame" default="camera_frame" />
    <arg name="image_topic" default="image_raw" />
  
    <!-- Apriltag algorithm -->
    <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="apriltag_node_$(arg camera_name)" clear_params="true" output="screen">
      <rosparam command="load" file="$(find apriltag_ros)/config/settings.yaml" />
      <rosparam command="load" file="$(find fav_object_detection)/config/tags_standalone.yaml" />
      <remap from="image_rect" to="$(arg camera_name)/$(arg image_topic)" />
      <param name="camera_frame" type="str" value="$(arg camera_frame)" />
      <param name="publish_tag_detections_image" type="bool" value="true" />
    </node>

    <node name="object_detection" pkg="fav_object_detection" type="object_detection.py" output="screen" >
      <param name="camera_name" value="$(arg camera_name)" />
    </node>
</launch>
